pkgs/development/python-modules/pybind11/default.nix

```
@@ -55,7 +54,7 @@ buildPythonPackage rec {
   checkInputs = [
     pytest
     numpy
-    scipy
+    # scipy not included due to recursive dependency
```

Perhaps this could work:
```suggestion
    (scipy.override { pybind11 = null; })
```